htdocs/xoops_lib/Xoops.php

```diff
@@ -733,7 +733,7 @@ public function isAdmin()
      *
      * @param boolean $optional true if failure to load handler should be considered a warning, not an error
      *
-     * @return \Xoops\Core\Kernel\Handlers\XoopsBlockHandler
+     * @return \Xoops\Core\Kernel\Handlers\XoopsBlockHandler|\Xoops\Core\Kernel\XoopsPersistableObjectHandler
```

Comment:
I know, PHPStorm loves to complain. I just don't always agree ;)

Sorry I wasn't clear. Right now, we have a set of factory methods calling a factory object. One too many factories involved is where the confusion sets in. 

That extra step is more of an artifact than a designed element. The behavior surrounding the kernel objects could be reconsidered with an eye toward simplification.